share/spack/gitlab/cloud_pipelines/stacks/ml-cpu/spack.yaml

```diff
@@ -0,0 +1,142 @@
+spack:
+  view: false
+
+  concretizer:
+    reuse: false
+    unify: false
+
+  config:
+    concretizer: clingo
+    install_tree:
+      root: /home/software/spack
+      padded_length: 384
+      projections:
+        all: "{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}"
+
+  packages:
+    all:
+      require:
+      - any_of: ["~cuda", "~rocm", "%gcc@11.2.0"]
```

Comment:
@alalazo this isn't working as expected, it's failing for packages that don't have a `+rocm` variant.